proposals/0051-options-local-pragma.rst

```diff
@@ -0,0 +1,145 @@
+Options Local Pragma
+====================
+
+.. proposal-number:: 
+.. ticket-url::
+.. implemented::
+.. highlight:: haskell
+.. header:: This proposal is `discussed at this pull request <https://github.com/ghc-proposals/ghc-proposals/pull/0>`_.
+.. sectnum::
+.. contents::
+
+We propose to add functionality for switching GHC-options locally. For instance, this code
+::
+ {-# OPTIONS_GHC -Wno-name-shadowing #-}
+ 
+ f = {-# OPTIONS_LOCAL -Wname-shadowing #-}
```

Comment:
This example doesn't demonstrate the local nature of `{-# OPTIONS_LOCAL #-}`. Perhaps add another function `g` with shadowing and say that the warning will be reported in `f` but not in `g`.

Reply:
```
f a =
  {-# OPTIONS_LOCAL -Wname-shadowing #-}
  \a -> a + a

g a = \a -> a + a
```